projects/plugins/jetpack/extensions/shared/components/product-management-controls/inspector-control.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	PanelBody,
+	PanelRow,
+	SelectControl,
+	TextControl,
+	ExternalLink,
+	Placeholder,
+	Spinner,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { lock } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { API_STATE_NOT_REQUESTING, API_STATE_REQUESTING } from './constants';
+import { CURRENCY_OPTIONS } from '../../currencies';
+import { store as membershipProductsStore } from '../../../store/membership-products';
+
+export default function ProductManagementInspectorControl( {
+	allowCreateOneTimeInterval,
+	setSelectedProductId,
+} ) {
+	const siteSlug = useSelect( select => select( membershipProductsStore ).getSiteSlug() );
+	const { saveProduct } = useDispatch( membershipProductsStore );
+
+	const [ apiState, setApiState ] = useState( API_STATE_NOT_REQUESTING );
+	const [ title, setTitle ] = useState( __( 'Monthly Subscription', 'jetpack' ) );
+	const [ currency, setCurrency ] = useState( 'USD' );
+	const [ price, setPrice ] = useState( 5 );
+	const [ interval, setInterval ] = useState( '1 month' );
+
+	const intervalOptions = [
+		{ label: __( 'Month', 'jetpack' ), value: '1 month' },
+		{ label: __( 'Year', 'jetpack' ), value: '1 year' },
+	];
+	if ( allowCreateOneTimeInterval ) {
+		intervalOptions.concat( [ { label: __( 'One-Time Payment', 'jetpack' ), value: 'one-time' } ] );
```

Comment:
We should update the array, the concat method returns a new array that is the combination of both. Since we aren't storing the new array we can never create "One-time Payments".